setup.py

```diff
@@ -105,6 +106,23 @@
 from tools.setup_helpers.dist_check import WITH_DISTRIBUTED, \
     WITH_DISTRIBUTED_MW, WITH_GLOO_IBVERBS
 
+################################################################################
+# Check Python build dependencies
+################################################################################
+missing_pydep = '''
+Missing build dependency: Unable to `import {importname}`.
+Please install it via `conda install {module}` or `pip install {module}`
+'''.strip()
+
+
+def check_pydep(importname, module):
+    try:
+        importlib.import_module(importname)
+    except ImportError:
+        raise RuntimeError(missing_pydep.format(importname=importname, module=module))
+
+check_pydep('yaml', 'pyyaml')
```

Comment:
Can we move this only inside `build`? `setup.py` has a ton of functions, and many of them likely don't require `pyyaml` (e.g. `egg-info`)